code/modules/mining/equipment/explorer_gear.dm

```diff
@@ -46,3 +46,93 @@
+
+/obj/item/clothing/suit/space/hostile_environment/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/space/hostile_environment/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/space/hostile_environment/process()
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
+		if(prob(2)) //cursed by bubblegum
+			if(prob(15))
+				new /datum/hallucination/oh_yeah(C)
+				to_chat(C, "<span class='colossus'><b>[pick("I AM IMMORTAL.","I SHALL TAKE BACK WHAT'S MINE.","I SEE YOU.","YOU CANNOT ESCAPE ME FOREVER.","DEATH CANNOT HOLD ME.")]</b></span>")
+			else
+				to_chat(C, "<span class='warning'>[pick("You hear faint whispers.","You smell ash.","You feel hot.","You hear a roar in the distance.")]</span>")
+
+/obj/item/clothing/suit/space/hostile_environment/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/toy/crayon/spraycan))
+		var/obj/item/toy/crayon/spraycan/spraycan = O
+		if(spraycan.is_capped)
+			to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+			return
+		if(spraycan.check_empty(user))
+			return
+		spraycan.use_charges(2)
+		add_atom_colour(spraycan.paint_color, FIXED_COLOUR_PRIORITY)
+		playsound(get_turf(user), 'sound/effects/spray.ogg', 5, 1, 5)
+		to_chat(user, "<span class='notice'>You sprays [spraycan] on [src], painting it.</span>")
+	else
+		return ..()
+
+/obj/item/clothing/head/helmet/space/hostile_environment
+	name = "H.E.C.K. helmet"
+	desc = "Hostile Environiment Cross-Kinetic Helmet: A helmet designed to withstand the wide variety of hazards from Lavaland. It wasn't enough for its last owner."
+	icon_state = "hostile_env"
+	item_state = "hostile_env"
+	w_class = WEIGHT_CLASS_NORMAL
+	max_heat_protection_temperature = FIRE_IMMUNITY_HELM_MAX_TEMP_PROTECT
+	flags_1 = THICKMATERIAL_1 // no space protection
+	armor = list(melee = 70, bullet = 40, laser = 10,energy = 10, bomb = 50, bio = 100, rad = 100, fire = 100, acid = 100)
+	resistance_flags = FIRE_PROOF | LAVA_PROOF
+
+/obj/item/clothing/head/helmet/space/hostile_environment/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/clothing/head/helmet/space/hostile_environment/update_icon()
+	..()
+	cut_overlays()
+	var/mutable_appearance/glass_overlay = mutable_appearance(icon, "hostile_env_glass")
+	glass_overlay.appearance_flags = RESET_COLOR
+	add_overlay(glass_overlay)
+
+/obj/item/clothing/head/helmet/space/hostile_environment/worn_overlays(isinhands)
+	. = ..()
+	if(!isinhands)
+		var/mutable_appearance/M = mutable_appearance('icons/mob/head.dmi', "hostile_env_glass")
+		M.appearance_flags = RESET_COLOR
+		. += M
+
+/obj/item/clothing/head/helmet/space/hostile_environment/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/toy/crayon/spraycan))
```

Comment:
Turn this behaviour into a component and add it to the suit and helmet instead of this attackby code

```dm
/datum/component/spraycan_paintable
    var/current_paint

/datum/component/spraycan_paintable/Initialize()
    RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/Repaint)

/datum/component/spraycan_paintable/Destroy()
    RemoveCurrentCoat()
    return ..()

/datum/component/spraycan_paintable/proc/RemoveCurrentCoat()
    var/atom/A = parent
    A.remove_atom_colour(current_paint, FIXED_COLOUR_PRIORITY)

/datum/component/spraycan_paintable/proc/Repaint(obj/item/toy/crayon/spraycan/spraycan, mob/living/user)
    if(!istype(spraycan) || user.a_intent == INTENT_HARM)
        return FALSE
    . = TRUE
    if(spraycan.is_capped)
        to_chat(user, "<span class='warning'>Take the cap off first!</span>")
        return
    if(spraycan.check_empty(user))
        return
    RemoveCurrentCoat()
    spraycan.use_charges(2)
    var/colour = current_paint = spraycan.paint_color
    var/atom/A = parent
    A.add_atom_colour(colour, FIXED_COLOUR_PRIORITY)
    playsound(spraycan, 'sound/effects/spray.ogg', 5, 1, 5)
    to_chat(user, "<span class='notice'>You spray [spraycan] on [A], painting it.</span>")
```